src/task-actions.ts

```diff
@@ -554,9 +554,14 @@ async function runWithDevnet(hre: HardhatRuntimeEnvironment, fn: () => Promise<u
         return;
     }
 
-    const port = await getFreePort();
-    const networkUrl = `http://127.0.0.1:${port}`;
-    hre.config.networks.integratedDevnet.url = networkUrl;
+    const { hostname, port } = new URL(hre.config.starknet.networkUrl);
+    const isPortFree = await isFreePort(parseInt(port));
+    if (!isPortFree) {
```

Comment:
Why this check? Couldn't it surprise the user if we just silently change the specified port. If the port is occupied, integrated-devnet spawner will inform the user about that, right?

My understanding was that we should find a free port only if the user didn't provide a URL? Now I'm actually thinking, is it even possible for the user to omit the URL? Can the type extensions file be modified to allow that?